// react-native-input-select/lib/commonjs/components/CheckBox/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _colors = require("../../styles/colors");

var _constants = require("../../constants");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/* eslint-disable react-native/no-inline-styles */
const CheckBox = _ref => {
  let {
    label,
    value,
    disabled,
    primaryColor,
    checkboxSize,
    checkboxStyle,
    checkboxLabelStyle,
    onChange
  } = _ref;
  const fillColor = {
    backgroundColor: disabled ? '#d3d3d3' : value ? (checkboxStyle === null || checkboxStyle === void 0 ? void 0 : checkboxStyle.backgroundColor) || primaryColor || 'green' : 'white',
    borderColor: disabled ? _colors.colors.disabled : styles.checkbox.borderColor
  };
  return /*#__PURE__*/_react.default.createElement(_reactNative.Pressable, {
    onPress: onChange ? () => onChange(!value) : null,
    style: [styles.checkboxContainer],
    disabled: disabled
  }, /*#__PURE__*/_react.default.createElement(_reactNative.View, {
    style: [styles.checkbox, checkboxStyle, fillColor]
  }, /*#__PURE__*/_react.default.createElement(_reactNative.Image, {
    source: require('../../asset/check.png'),
    style: [{
      height: checkboxSize || _constants.CHECKBOX_SIZE,
      width: checkboxSize || _constants.CHECKBOX_SIZE
    }]
  })), label && /*#__PURE__*/_react.default.createElement(_reactNative.Text, {
    style: [checkboxLabelStyle, styles.labelStyle]
  }, label));
};

const styles = _reactNative.StyleSheet.create({
  checkboxContainer: {
    flexDirection: 'row',
    flexWrap: 'nowrap',
    alignItems: 'center'
  },
  checkbox: {
    padding: 4,
    borderWidth: 1,
    borderStyle: 'solid',
    borderRadius: 4,
    borderColor: 'black'
  },
  labelStyle: {
    marginLeft: 10
  }
});

var _default = CheckBox;
exports.default = _default;
