// react-native-input-select/lib/module/components/Dropdown/DropdownListItem.js
import React, { memo } from 'react';
import { TouchableOpacity, StyleSheet } from 'react-native';
import CheckBox from '../CheckBox';

const DropdownListItem = _ref => {
  let {
    item,
    optionLabel,
    optionValue,
    isMultiple,
    selectedOption,
    onChange,
    primaryColor,
    checkboxSize,
    checkboxStyle,
    checkboxLabelStyle
  } = _ref;
  const selectedOptionValue = optionValue !== null && optionValue !== void 0 ? optionValue : 'value';
  return /*#__PURE__*/React.createElement(TouchableOpacity, {
    style: styles.dropdownModalOptions,
    onPress: item.disabled ? () => {} : () => onChange(item[selectedOptionValue]) // intentionally didn't use the disable property

  }, /*#__PURE__*/React.createElement(CheckBox, {
    value: isMultiple ? selectedOption.includes(item[selectedOptionValue]) : [selectedOption].includes(item[selectedOptionValue]),
    label: item[optionLabel !== null && optionLabel !== void 0 ? optionLabel : ''],
    onChange: () => onChange(item[selectedOptionValue]),
    primaryColor: primaryColor,
    checkboxSize: checkboxSize,
    checkboxStyle: checkboxStyle,
    checkboxLabelStyle: checkboxLabelStyle,
    disabled: item.disabled
  }));
};

const styles = StyleSheet.create({
  dropdownModalOptions: {
    paddingHorizontal: 20,
    paddingVertical: 10,
    flexDirection: 'row',
    alignItems: 'center'
  }
});
export default /*#__PURE__*/memo(DropdownListItem);
