// react-native-input-select/lib/module/components/Dropdown/DropdownSelectedItemsView.js
import React from 'react';
import { View, Text, Pressable, ScrollView, StyleSheet, Image, TouchableOpacity } from 'react-native';
import { colors } from '../../styles/colors';
import { inputStyles } from '../../styles/input';

const DropdownSelectedItemsView = _ref => {
  var _getSelectedItemsLabe;

  let {
    placeholder,
    error,
    getSelectedItemsLabel,
    handleToggleModal,
    isMultiple,
    selectedItem,
    selectedItems,
    dropdownIcon,
    dropdownStyle,
    dropdownIconStyle,
    selectedItemStyle,
    placeholderStyle,
    multipleSelectedItemStyle,
    dropdownErrorStyle,
    primaryColor,
    disabled
  } = _ref;
  return /*#__PURE__*/React.createElement(Pressable, {
    onPress: () => handleToggleModal(),
    style: _ref2 => {
      let {
        pressed
      } = _ref2;
      return [pressed && { ...inputStyles.inputFocusState,
        borderColor: primaryColor
      }, inputStyles.input, dropdownStyle, error && //this must be last
      error !== '' && !pressed && { ...inputStyles.inputFocusErrorState,
        ...dropdownErrorStyle
      }];
    },
    disabled: disabled
  }, /*#__PURE__*/React.createElement(ScrollView, {
    horizontal: true,
    alwaysBounceHorizontal: true,
    showsHorizontalScrollIndicator: false
  }, /*#__PURE__*/React.createElement(View, {
    style: styles.selectedItemsContainer,
    onStartShouldSetResponder: () => true
  }, isMultiple ? (_getSelectedItemsLabe = getSelectedItemsLabel()) === null || _getSelectedItemsLabe === void 0 ? void 0 : _getSelectedItemsLabe.map((item, i) => /*#__PURE__*/React.createElement(TouchableOpacity, {
    onPress: () => handleToggleModal(),
    key: `react-native-input-select-${Math.random()}-${i}`,
    disabled: disabled
  }, /*#__PURE__*/React.createElement(Text, {
    style: [styles.selectedItems, {
      backgroundColor: primaryColor
    }, multipleSelectedItemStyle]
  }, item))) : /*#__PURE__*/React.createElement(TouchableOpacity, {
    onPress: () => handleToggleModal(),
    disabled: disabled
  }, /*#__PURE__*/React.createElement(Text, {
    style: [styles.blackText, selectedItemStyle]
  }, getSelectedItemsLabel()))), !selectedItem && (selectedItems === null || selectedItems === void 0 ? void 0 : selectedItems.length) === 0 && /*#__PURE__*/React.createElement(Text, {
    style: [styles.blackText, placeholderStyle]
  }, placeholder !== null && placeholder !== void 0 ? placeholder : 'Select an option')), /*#__PURE__*/React.createElement(View, {
    style: [styles.iconStyle, dropdownIconStyle]
  }, dropdownIcon || /*#__PURE__*/React.createElement(Image, {
    source: require('../../asset/arrow-down.png')
  })));
};

const styles = StyleSheet.create({
  iconStyle: {
    position: 'absolute',
    right: 25,
    top: 25
  },
  selectedItemsContainer: {
    flexDirection: 'row',
    flexWrap: 'nowrap',
    alignItems: 'center'
  },
  selectedItems: {
    color: colors.white,
    paddingHorizontal: 10,
    paddingVertical: 5,
    borderRadius: 10,
    backgroundColor: colors.primary,
    marginRight: 10,
    overflow: 'hidden'
  },
  blackText: {
    color: colors.black
  }
});
export default DropdownSelectedItemsView;
