// react-native-input-select/lib/commonjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.DropdownSelect = void 0;

var _react = _interopRequireWildcard(require("react"));

var _reactNative = require("react-native");

var _Dropdown = _interopRequireDefault(require("./components/Dropdown/Dropdown"));

var _DropdownList = _interopRequireDefault(require("./components/Dropdown/DropdownList"));

var _CustomModal = _interopRequireDefault(require("./components/CustomModal"));

var _Input = require("./components/Input");

var _CheckBox = _interopRequireDefault(require("./components/CheckBox"));

var _colors = require("./styles/colors");

var _constants = require("./constants");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

const DropdownSelect = _ref => {
  let {
    placeholder,
    label,
    error,
    helperText,
    options,
    optionLabel,
    optionValue,
    onValueChange,
    selectedValue,
    isMultiple,
    isSearchable,
    dropdownIcon,
    labelStyle,
    placeholderStyle,
    dropdownStyle,
    dropdownIconStyle,
    dropdownContainerStyle,
    dropdownErrorStyle,
    dropdownErrorTextStyle,
    dropdownHelperTextStyle,
    selectedItemStyle,
    multipleSelectedItemStyle,
    modalBackgroundStyle,
    modalOptionsContainerStyle,
    searchInputStyle,
    primaryColor,
    disabled,
    checkboxSize,
    checkboxStyle,
    checkboxLabelStyle,
    listHeaderComponent,
    listFooterComponent,
    modalProps,
    hideModal = false,
    ...rest
  } = _ref;
  const [newOptions, setNewOptions] = (0, _react.useState)([]);
  const [open, setOpen] = (0, _react.useState)(false);
  const [selectAll, setSelectAll] = (0, _react.useState)(false);
  const [selectedItem, setSelectedItem] = (0, _react.useState)(''); //for single selection

  const [selectedItems, setSelectedItems] = (0, _react.useState)([]); //for multiple selection

  const [searchValue, setSearchValue] = (0, _react.useState)('');
  (0, _react.useEffect)(() => {
    if (options) {
      setNewOptions(options);
    }
  }, [options]);
  (0, _react.useEffect)(() => {
    isMultiple ? setSelectedItems(Array.isArray(selectedValue) ? selectedValue : []) : setSelectedItem(selectedValue);
  }, [selectedValue, isMultiple, onValueChange]);
  /*===========================================
   * Selection handlers
   *==========================================*/

  const handleSingleSelection = value => {
    if (selectedItem === value) {
      setSelectedItem(null);
      onValueChange(null); //send value to parent
    } else {
      setSelectedItem(value);
      onValueChange(value); //send value to parent

      setOpen(false); //close modal upon selection
    }
  };

  const handleMultipleSelections = value => {
    setSelectedItems(prevVal => {
      var _selectedValues;

      let selectedValues = [...prevVal];

      if ((_selectedValues = selectedValues) !== null && _selectedValues !== void 0 && _selectedValues.includes(value)) {
        selectedValues = selectedValues.filter(item => item !== value);
      } else {
        selectedValues.push(value);
      }

      setSelectedItems(selectedValues);
      onValueChange(selectedValues); //send value to parent
      //select all checkbox should not be checked if the list contains disabled values

      if (options.filter(item => !item.disabled).length === selectedValues.length) {
        setSelectAll(true);
      } else {
        setSelectAll(false);
      }

      return selectedValues;
    });
  };

  const handleSelectAll = () => {
    setSelectAll(prevVal => {
      const selectedValues = [];
      const filteredOptions = newOptions.filter(item => !item.disabled); //don't select disabled items

      if (!prevVal) {
        for (let i = 0; i < filteredOptions.length; i++) {
          selectedValues.push(filteredOptions[i][optionValue]);
        }
      }

      setSelectedItems(selectedValues);
      onValueChange(selectedValues); //send value to parent

      return !prevVal;
    });
  };
  /*===========================================
   * Get label handler
   *==========================================*/


  const getSelectedItemsLabel = () => {
    if (isMultiple && Array.isArray(selectedItems)) {
      let selectedLabels = [];
      selectedItems === null || selectedItems === void 0 ? void 0 : selectedItems.forEach(element => {
        var _options$find;

        let selectedItemLabel = options && ((_options$find = options.find(item => item[optionValue !== null && optionValue !== void 0 ? optionValue : _constants.DEFAULT_OPTION_VALUE] === element)) === null || _options$find === void 0 ? void 0 : _options$find[optionLabel]);
        selectedLabels.push(selectedItemLabel);
      });
      return selectedLabels;
    }

    let selectedItemLabel = options && options.find(item => item[optionValue !== null && optionValue !== void 0 ? optionValue : _constants.DEFAULT_OPTION_VALUE] === selectedItem);
    return selectedItemLabel === null || selectedItemLabel === void 0 ? void 0 : selectedItemLabel[optionLabel !== null && optionLabel !== void 0 ? optionLabel : _constants.DEFAULT_OPTION_LABEL];
  };
  /*===========================================
   * Search
   *==========================================*/


  const onSearch = value => {
    setSearchValue(value);
    let searchText = value.toString().toLocaleLowerCase().trim();
    const regexFilter = new RegExp(searchText, 'i');
    const searchResults = options.filter(item => {
      if (item[optionLabel !== null && optionLabel !== void 0 ? optionLabel : _constants.DEFAULT_OPTION_LABEL].toString().toLowerCase().search(regexFilter) !== -1 || item[optionValue !== null && optionValue !== void 0 ? optionValue : _constants.DEFAULT_OPTION_VALUE].toString(regexFilter).toLowerCase().search(regexFilter) !== -1) {
        return item;
      }
    });
    setNewOptions(searchResults);
  };
  /*===========================================
   * Modal
   *==========================================*/


  const handleToggleModal = () => {
    setOpen(!open);
    setSearchValue('');
    setNewOptions(options);
  };

  (0, _react.useEffect)(() => {
    if (hideModal) {
      setOpen(false);
    }
  }, [hideModal]);
  let primary = primaryColor || _colors.colors.gray;
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_Dropdown.default, _extends({
    label: label,
    placeholder: placeholder,
    helperText: helperText,
    error: error,
    getSelectedItemsLabel: getSelectedItemsLabel,
    selectedItem: selectedItem,
    selectedItems: selectedItems,
    handleToggleModal: handleToggleModal,
    labelStyle: labelStyle,
    dropdownIcon: dropdownIcon,
    dropdownStyle: dropdownStyle,
    dropdownIconStyle: dropdownIconStyle,
    dropdownContainerStyle: dropdownContainerStyle,
    dropdownErrorStyle: dropdownErrorStyle,
    dropdownErrorTextStyle: dropdownErrorTextStyle,
    dropdownHelperTextStyle: dropdownHelperTextStyle,
    selectedItemStyle: selectedItemStyle,
    multipleSelectedItemStyle: multipleSelectedItemStyle,
    isMultiple: isMultiple,
    primaryColor: primary,
    disabled: disabled,
    placeholderStyle: placeholderStyle
  }, rest)), /*#__PURE__*/_react.default.createElement(_CustomModal.default, {
    open: open,
    handleToggleModal: handleToggleModal,
    modalBackgroundStyle: modalBackgroundStyle,
    modalOptionsContainerStyle: modalOptionsContainerStyle,
    onRequestClose: () => {},
    modalProps: modalProps
  }, /*#__PURE__*/_react.default.createElement(_DropdownList.default, {
    ListHeaderComponent: /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, isSearchable && /*#__PURE__*/_react.default.createElement(_Input.Input, {
      value: searchValue,
      onChangeText: text => onSearch(text),
      style: searchInputStyle,
      primaryColor: primary
    }), listHeaderComponent, isMultiple && newOptions.length > 1 && /*#__PURE__*/_react.default.createElement(_reactNative.View, {
      style: styles.optionsContainerStyle
    }, /*#__PURE__*/_react.default.createElement(_reactNative.TouchableOpacity, {
      onPress: () => {}
    }, /*#__PURE__*/_react.default.createElement(_CheckBox.default, {
      value: selectAll,
      label: selectAll ? 'Clear all' : 'Select all',
      onChange: () => handleSelectAll(),
      primaryColor: primary,
      checkboxSize: checkboxSize,
      checkboxStyle: checkboxStyle,
      checkboxLabelStyle: checkboxLabelStyle
    })))),
    ListFooterComponent: listFooterComponent,
    options: newOptions,
    optionLabel: optionLabel,
    optionValue: optionValue,
    isMultiple: isMultiple,
    isSearchable: isSearchable,
    selectedItems: selectedItems,
    selectedItem: selectedItem,
    handleMultipleSelections: handleMultipleSelections,
    handleSingleSelection: handleSingleSelection,
    primaryColor: primary,
    checkboxSize: checkboxSize,
    checkboxStyle: checkboxStyle,
    checkboxLabelStyle: checkboxLabelStyle
  })));
};

exports.DropdownSelect = DropdownSelect;

const styles = _reactNative.StyleSheet.create({
  optionsContainerStyle: {
    paddingHorizontal: 20,
    paddingVertical: 10,
    flexDirection: 'row'
  }
});

var _default = DropdownSelect;
exports.default = _default;
