// react-native-input-select/lib/module/components/CheckBox/index.js
/* eslint-disable react-native/no-inline-styles */
import React from 'react';
import { Pressable, Text, StyleSheet, Image, View } from 'react-native';
import { colors } from '../../styles/colors';
import { CHECKBOX_SIZE } from '../../constants';

const CheckBox = _ref => {
  let {
    label,
    value,
    disabled,
    primaryColor,
    checkboxSize,
    checkboxStyle,
    checkboxLabelStyle,
    onChange
  } = _ref;
  const fillColor = {
    backgroundColor: disabled ? '#d3d3d3' : value ? (checkboxStyle === null || checkboxStyle === void 0 ? void 0 : checkboxStyle.backgroundColor) || primaryColor || 'green' : 'white',
    borderColor: disabled ? colors.disabled : styles.checkbox.borderColor
  };
  return /*#__PURE__*/React.createElement(Pressable, {
    onPress: onChange ? () => onChange(!value) : null,
    style: [styles.checkboxContainer],
    disabled: disabled
  }, /*#__PURE__*/React.createElement(View, {
    style: [styles.checkbox, checkboxStyle, fillColor]
  }, /*#__PURE__*/React.createElement(Image, {
    source: require('../../asset/check.png'),
    style: [{
      height: checkboxSize || CHECKBOX_SIZE,
      width: checkboxSize || CHECKBOX_SIZE
    }]
  })), label && /*#__PURE__*/React.createElement(Text, {
    style: [checkboxLabelStyle, styles.labelStyle]
  }, label));
};

const styles = StyleSheet.create({
  checkboxContainer: {
    flexDirection: 'row',
    flexWrap: 'nowrap',
    alignItems: 'center'
  },
  checkbox: {
    padding: 4,
    borderWidth: 1,
    borderStyle: 'solid',
    borderRadius: 4,
    borderColor: 'black'
  },
  labelStyle: {
    marginLeft: 10
  }
});
export default CheckBox;
